.circleci/config.yml

```diff
@@ -853,6 +863,72 @@ jobs:
           command: |
             ./script/chart_upstream_checker.sh << pipeline.parameters.CI_BOT_USERNAME >> << pipeline.parameters.CI_BOT_EMAIL >> << pipeline.parameters.CI_BOT_GPG >> << pipeline.parameters.CI_BOT_FORKED_CHARTS_DEPLOYKEY_FILENAME >> << pipeline.parameters.CHARTS_REPO_ORIGINAL >> << pipeline.parameters.BRANCH_CHARTS_REPO_ORIGINAL >> << pipeline.parameters.CHARTS_REPO_FORKED >> << pipeline.parameters.BRANCH_CHARTS_REPO_FORKED >> << pipeline.parameters.KUBEAPPS_REPO >> << pipeline.parameters.BRANCH_KUBEAPPS_REPO >> << pipeline.parameters.README_GENERATOR_REPO >>
+  report_srp:
+    environment:
+      <<: *common_envars
+    docker:
+      - image: cimg/go:<< pipeline.parameters.GOLANG_VERSION >>
+        auth:
+          username: $DOCKER_USERNAME
+          password: $DOCKER_PASSWORD
+    steps:
+      - checkout
+      - run:
+          name: Init submodules
+          command: |
+            git submodule update --init --recursive
+      - run:
+          name: Install and configure the SRP tool
+          command: |
+            wget https://vmwaresaas.jfrog.io/artifactory/srp-tools/srpcli/<< pipeline.parameters.SRP_VERSION >>/linux/srp
+            sudo chmod +x srp
+            sudo mv ./srp /usr/local/bin/
+            srp --version
+            srp config --srp-endpoint << pipeline.parameters.SRP_ENDPOINT >>
+            srp config auth --client-id $KUBEAPPS_SRP_CLIENT_ID --client-secret $KUBEAPPS_SRP_CLIENT_SECRET
+      # - run:
+      #     name: Install and configure the SRP observer tool
+      #     command: |
+      #       wget https://s3.amazonaws.com/srp-cli/linux-observer-latest.tar.gz
+      #       tar zxf linux-observer-latest.tar.gz
+      - run:
+          name: Generate the "source provenance" file
+          command: |
+            # This UID should have been already set in the internal tool.
+            PROV_OBJ_UID="uid.obj.build.circleci(instance='circleci.com',type='<< pipeline.project.type >>',namespace='${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}',branch='${CIRCLE_BRANCH}',commit_id='${CIRCLE_SHA1}',build_id='<< pipeline.number >>/${CIRCLE_WORKFLOW_ID}/${CIRCLE_BUILD_NUM}')"
+            echo "PROV_OBJ_UID=${PROV_OBJ_UID}"
+            # Using CIRCLE_TAG as the version if defined (i.e., when releasing a version); use the commit SHA otherwise.
+            VERSION=$([ -z "${CIRCLE_TAG}" ] && echo "${CIRCLE_SHA1}" || echo "${CIRCLE_TAG}")
+            srp provenance source --verbose --scm-type git --name "kubeapps" --path ./ --saveto ./source-provenance.json --comp-uid "${PROV_OBJ_UID}" --build-number "${CIRCLE_WORKFLOW_ID}" --version "${VERSION}" --all-ephemeral true --build-type release
+            cat ./source-provenance.json
+      - store_artifacts:
+          path: source-provenance.json
+      # TODO(agamez): we will need to create the "network provenance" file soon.
+      # - run:
+      #     name: Generate the "network provenance" file
```

Comment:
We may have to ask for a reporter for Rust, I haven't been able to make it work. However, it is not a priority right now. The provenance file is enough.